contributors/design-proposals/scheduling/scheduling-policy.md

```diff
@@ -0,0 +1,781 @@
+```
+
+
+#### allowed
+Allow pods to use either the green-scheduler (which is used by specifying `spec.schedulerName` in pod definition) or the `my-scheduler` scheduler (by specifying `spec.schedulerName: "my-scheduler"`):
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    SchedulerNames: ["green-scheduler", "my-scheduler"]
+```
+
+An empty list of schedulerNames will allow usage of all schedulers:
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    SchedulerNames: []
+```
+
+#### default
+pods will default use either the my-scheduler if nothing is specified in `spec.schedulerName`:
+```yaml
+Kind: SchedulingPolicy
+spec:
+  default:
+    SchedulerName: "my-scheduler"
+```
+
+
+### Tolerations
+
+Toleration usage can be regulated using fine-grain rules with `tolerations` field. If specifying multiple `tolerations`, pod will be scheduled if one of the tolerations is satisfied.
+
+
+#### Allowed
+
+This allows requires tolerations in the following forms:
+- tolerations that tolerates taints with key named `mykey` that has a value `value` and with a `NoSchedule` effect.
+- tolerations that tolerates taints with key `other_key` that has a `NoExecute` effect.
+
+##### Fine-grain allowed tolerations
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    tolerations:
+    - keys: ["mykey"]
+      operators: ["Equal"]
+      values: ["value"]
+      effects: ["NoSchedule"]
+    - keys: ["other_key"]
+      operators: ["Exists"]
+      effects: ["NoExecute"]
+```
+
+Here we allow tolerations in the following forms:
+- tolerations that tolerates all `PreferNoSchedule` taints with any value.
+- tolerations that tolerates taints based on any key existence with effect `NoSchedule`.
+Also note that this SchedulingPolicy does not allow tolerating NoExecute taints.
+
+##### Coarse-grain allowed tolerations
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    tolerations:
+    - keys: []
+      operators: []
+      values: []
+      effects: ["PreferNoSchedule"]
+    - keys: []
+      operators: ["Exists"]
+      effects: ["NoSchedule"]
+```
+an empty list of toleration allows all types of tolerations:
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    tolerations: []
+```
+
+Which is equivalent to:
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    tolerations:
+    - keys: []
+      operators: []
+      values: []
+      effects: []
+```
+
+#### default
+
+if no toleration is not specified, the following `SchedulingPolicy` will add a:
+- toleration for <mykey,value,NoSchedule> and <mykey,other_value,NoSchedule> Taints.
+- toleration for <other_key,NoExecute> Taint.
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  default:
+    tolerations:
+    - key: "mykey"
+      operator: "Equal"
+      values: ["value","other_value"]
+      effect: "NoSchedule"
+    - key: "other_key"
+      operator: "Exists"
+      effect: "NoExecute"
+```
+note: an empty array of toleration is not a valid syntax for default toleration.
```

Comment:
This effectively means that with SchedulingPolicy enabled, every pod must have a toleration.

Reply:
Can tolerations not be specified at all?